### DISCUSSION TITLE
Pages X Exercises

### DISCUSSION THREAD

**Discussion by nimonian** [corndeladmin/tech-docs#2] [General]
I just wanted to make space to document our discussion about how pages are arranged, and how they link to exercises.

How do we feel about the [Array Methods](https://github.com/corndeladmin/tech-docs/blob/main/src/js/array-methods.md) page?

  - Is it logical enough?
  - Is it versatile enough to be used by other programmes?
  - Does it feel bitesize enough?

For example, would separate pages for
  - push, pop
  - map
  - filter

be desirable?

**Comment by nimonian** [comment]
Part of why I ask is, having looked at how W3 schools organise their content, it's a lot more broken down, and perhaps feels a bit more approachable

But then they have more clicking to do to move around the pages, so there's pros and cons

**Reply by AndrewYHuang** [reply]
I'm not convinced that the docs on W3s are that much more broken down, having looked at their [array methods page](https://www.w3schools.com/js/js_array_methods.asp). Each section on a method is a lot more terse, which works well because of the `Try it Yourself` button.

**Reply by nimonian** [reply]
You're absolutely right - I think I was looking at a different set of pages last night which were each much shorter, and I didn't find a page of this length

I also think the "on this page" nav we have on the right provides a good overview and makes the page feel more manageable/navigable

![image](https://github.com/corndeladmin/tech-docs/assets/12779491/60e3015d-67ce-4b70-ac13-c4e0695c82b2)

**Reply by AndrewYHuang** [reply]
That's a great point! I was wondering what helped me navigate the w3s page more than the raw markdown file, and I noticed I was using the ToC at the top. The nav on the right serves the same purpose and helps chunk down the page given that we have meaningful headings
